### ISSUE TITLE
Virtual dom re-renders unchanged node

### ISSUE THREAD

**Issue by groz** [elm-lang/virtual-dom#64]
Seeing an odd behavior going through [HTTP EXAMPLE](http://elm-lang.org/examples/http) in Elm guide. 
Every time I click `More Please!` the current gif resets before an update.

Seems like this is what happening:
- `onClick` produces `MorePlease` message
- `update` receives `MorePlease` and returns `(model, getRandomGif model.topic)`, so it doesn't mutate `model`
- `view` should return unchanged `VirtualDom.Node` compared to the previous step
- but `img` tag gets re-rendered anyway so gif gets reset

Is it a bug in VirtualDom renderer? I thought it wasn't supposed to re-render elements if the model didn't change

OS: macOS Sierra Version 10.12.1 (16B2555)
Browser: Google Chrome Version 54.0.2840.98 (64-bit)
Elm: 0.18

**Comment by peteygao** [comment]
**tl;dr** This is **not** a `elm-lang/virtual-dom` bug!

### Explanation
This seems to be a **browser renderer bug** that Blink inherited from Webkit, as I can confirm the 
same issue exists in Safari.

What's happening is that the Node's `src` attribute is changed upon the `More Please!` button click (well, it changes _very quickly_ after the click. Within <50ms), so it looks like the trigger is the button click, when in fact it's the `src` attribute being updated to the URL of the next GIF image.

Now the bug in Webkit/Blink is that the current GIF is reset when the `src` is changed to the new GIF, even though the new GIF hasn't loaded yet.

You will not see this problem in IE, Edge nor FF (tested with IE10.0.9200.17609, Edge 38.14393.0.0, and FF 53.0a2) as they do not reset the _current_ GIF when the `src` changes (Edge will stutter and pause the current GIF as it attempts to load the next GIF, however).

If you want to reproduce this "bug", enable the Network Throttling in Chrome:
![screen shot 2017-03-06 at 3 01 19 pm](https://cloud.githubusercontent.com/assets/2184561/23634314/715da23e-027e-11e7-8b80-3f09763789d8.png)

Make sure you select a throttling profile with a large amount of latency (in this case, 500ms). Now, inspect the `<img>` element to obtain focus on it, and then click on the `More Please!` button. After ~500ms, you should see the `src` attribute change, and _then_ the GIF will reset. Proof that the virtual-dom is doing its job properly 😄.

I briefly searched the Chromium bug tracker and didn't see a ticket for this behaviour. You could be the first to report this problem on their bug tracker and win some virtual brownie points!

**Comment by evancz** [comment]
Based on @peteygao's analysis, it looks like this is not something we can address in our library. So I will close.

Thanks for the report @groz, and thanks @peteygao for doing a bunch of great research on what is going on!
